### PR TITLE
refactor: reduce number of args for `post_block_balance_increments`

### PR DESCRIPTION
**Pull request by tcoratger**

### CODE REVIEW

File: crates/revm/src/state_change.rs [hunk 1]
```diff
@@ -36,40 +36,34 @@ use std::collections::HashMap;
...
 /// state changes (DAO fork).
-#[allow(clippy::too_many_arguments)]
 #[inline]
 pub fn post_block_balance_increments(
     chain_spec: &ChainSpec,
-    block_number: u64,
-    block_difficulty: U256,
-    beneficiary: Address,
-    block_timestamp: u64,
+    block: &BlockWithSenders,
```
Comment (mattsse) [inline crates/revm/src/state_change.rs:42]: I don't think we need the senders here, right?

can we just accept `&Block` via deref:

https://github.com/paradigmxyz/reth/blob/9fd2cf027f130e14d75ca6eecb24b13ab6dec1ba/crates/primitives/src/block.rs#L198-L202

**Review by mattsse — CHANGES_REQUESTED** [review]
this could be slightly problematic in the near future when we consider different block types, although this should then be solved with a trait instead. because this is a byproduct of the block, makes more sense to provide the block as the argument
and I agree this change makes it a lot easier to read

**Comment by tcoratger** [comment]
> this could be slightly problematic in the near future when we consider different block types, although this should then be solved with a trait instead. because this is a byproduct of the block, makes more sense to provide the block as the argument and I agree this change makes it a lot easier to read

yeah it was with the aim of progressive change, this was the first step towards more abstraction imo

File: crates/revm/src/state_change.rs (hunk 1 above)
Comment (tcoratger) [reply crates/revm/src/state_change.rs]: fixed

**Review by mattsse — APPROVED** [review]

**Pull request merged**